govcd/externalnetwork_test.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	. "gopkg.in/check.v1"
+	"net/url"
+	"time"
+)
+
+// Retrieves an external network and checks that its contents are filled as expected
+func (vcd *TestVCD) Test_ExternalNetworkGetByName(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+
+	if vcd.config.VCD.ExternalNetwork == "" {
+		check.Skip("Test_GetByName: External network isn't configured. Test can't proceed")
+	}
+
+	externalNetwork := NewExternalNetwork(&vcd.client.Client)
+	err := externalNetwork.GetByName(vcd.config.VCD.ExternalNetwork)
+	check.Assert(err, IsNil)
+
+	check.Assert(externalNetwork.ExternalNetwork.Name, Equals, vcd.config.VCD.ExternalNetwork)
+}
+
+// Tests System function Delete by creating external network and
```

Comment:
A few problems with this test:

1. We need a check that the needed elements were set in the configuration, and abort if they weren't.
2. If the Port group type is wrong, the test fails with a message saying that the port group was not found, instead of saying that the type doesn't match. Is there a way of giving a better message?

NOTE ON TESTING: if we have an already configured Cassini vCD, we can't test the NETWORK type, as it's already taken. To test the creation of a network (using `VM Network`, for example), we need to define at least an organization, and run the test with environment variable `GOVCD_SKIP_VAPP_CREATION=1`.

Reply:
1. I added checks in tests to abort when configuration is missing.
2. This is query vDC by name or type. Can be anything between so no good option to tell what is wrong automatically.